### PR TITLE
SRE-9 Secrets Detection for Helm Values

### PR DESCRIPTION
**Pull request by cemery93**
This adds in a hook to provide scanning of helm values files for detection of plaintext credentials. See https://linear.app/plentiau/issue/SRE-9/linting-and-checks-for-k8s-repos for more details and context.

### CODE REVIEW

**Comment by cemery93** [comment]
Example output with redacted output
image

**Comment by cemery93** [comment]
Output running as a pre-commit hook on another repo
image

File: plaintext-secrets [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+declare failedFilesCount=0
+declare return_code=0
+
+# Iterate over files matching the naming pattern "helm-values.yaml"
+while IFS= read -r file; do
+  # Parse the values file and detect non-empty string values for keys named "password"
+  plaintext_password=$(yq eval '.. | select(has("password") and .password != "" and (.password | type == "!!str") and (.password | line_comment | . != "ignore-plaintext-pwd-check")) | .password' "$file")
+  plaintext_rootPassword=$(yq eval '.. | select(has("rootPassword") and .rootPassword != "" and (.rootPassword | type == "!!str") and (.rootPassword | line_comment | . != "ignore-plaintext-pwd-check")) | .rootPassword' "$file")
+  plaintext_adminPassword=$(yq eval '.. | select(has("adminPassword") and .adminPassword != "" and (.adminPassword | type == "!!str") and (.adminPassword | line_comment | . != "ignore-plaintext-pwd-check")) | .adminPassword' "$file")
+  plaintext_postgresqlPassword=$(yq eval '.. | select(has("postgresqlPassword") and .postgresqlPassword != "" and (.postgresqlPassword | type == "!!str") and (.postgresqlPassword | line_comment | . != "ignore-plaintext-pwd-check")) | .postgresqlPassword' "$file")
```
Comment (AlexeyPetroff) [inline plaintext-secrets]: Feel like its repetative. Could we consider having an array to store keys ("password" "rootPassword" "adminPassword" "postgresqlPassword") and iterate over them?

File: plaintext-secrets [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  else
+    ((passedFilesCount++))
+  fi
+done <<< "$(find . -type f -name "*helm-values*.yaml")"
+
+echo -e "\e[32mPassed Files Count: $passedFilesCount\e[0m"
+echo -e "\e[31mFailed Files Count: $failedFilesCount\e[0m"
+
+# Update return code to error if there were potential credentials found
+if [[ $failedFilesCount > 0 ]]; then
```
Comment (AlexeyPetroff) [inline plaintext-secrets:43]: My personal preference `if (( failedFilesCount > 0 ));`

**Review by AlexeyPetroff — APPROVED** [review]

File: plaintext-secrets (hunk 1 above)
Comment (cemery93) [reply plaintext-secrets]: I did try an array to begin with, but the issue then is if any of those password types returns more than one value for a file, we'd need a multi-dimensional array and bash really doesn't handle those well. If it starts becoming an issue, we could always rewrite the script into python to get better support for it

Comment (cemery93) [reply plaintext-secrets]: Fixed in latest commit

**Pull request merged**